### PR TITLE
API Controller Working with DB

### PR DESCRIPTION
**Pull request by hmitrea**
- Added 2 new controllers: API Controller & User Controller
- Added the DB
- Twitter OAuth Working
- API Route

### CODE REVIEW

File: server/controllers/apiControllers.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+const apiController = {};
+
+apiController.home = async (req, res, next)=>{
+    // will have no req body
+    // will make a request from the api
+    // saving it to res.locals
+    //sending resonse onwards
+    // https://api.data.charitynavigator.org/v2/Organizations?app_id=acad48f7&app_key=1caab9869fdb918b252c8f56de6b62ce&pageSize=999&sort=RATING
+    // If no requests have been made display the fetch
```
Comment (simple-sifu) [inline server/controllers/apiControllers.js:13]: Do you still need this url? or can we delete it?

File: server/controllers/apiControllers.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    .catch(err => console.log('The error when fetching the home API is ', err));
+    return next()
+}
+
+apiController.save = (req, res, next)=>{
+    // will post req.body
+    console.log(req.query)
+    let { name, financialrating, accountabilityrating, category, href, id} = req.query;
+    //!! make sure to tell tommy to give null for charities without category and href and 0 for charities without ratings
+    // send it to the db
```
Comment (simple-sifu) [inline server/controllers/apiControllers.js:29]: OK...I will try to remember...ha ha

File: server/models/database.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+//     category varchar, 
+//     href varchar);
+
+// CREATE TABLE IF NOT EXISTS charity_favs (
+//     _id SERIAL PRIMARY KEY, 
+//     user_id int, 
+//     charity_id int, 
+//     FOREIGN KEY (user_id) REFERENCES users(_id), 
+//     FOREIGN KEY (charity_id) REFERENCES charity(_id));
+
```
Comment (simple-sifu) [inline server/models/database.js:29]: can we remove it?

File: server/controllers/apiControllers.js (hunk 2 above)
Comment (hmitrea) [reply server/controllers/apiControllers.js]: Perfect

**Pull request merged**